### DISCUSSION TITLE
Are there any papers / references on SACI still available?

### DISCUSSION THREAD

**Discussion by mindcrime** [jason-lang/jason#116] [Q&A] [answered]
Hi folks, quick question. I was trying to find more details on SACI, and have run into a bit of a dead-end. All the references to SACI I've found online point to http://www.lti.pcs.usp.br/saci/ which just returns a 404. And even the Internet Archive "Wayback Machine" does not seem to have anything archived from that site.

Are there any papers, manuals, etc. related to SACI that some of you might have cached locally, or know of a copy online, that you could direct me to? It would be much appreciated. 

EDIT: I found "SACI: Uma ferramenta para implementação e monitoração da comunicação entre agentes" and Jomi's list of publications, which is awesome. It looks like there was (or is) a "SACI Programmer's Guide"? Any chance anybody knows where to find a copy of that?

I know this probably isn't terribly important to modern Jason, but I'm a big fan of studying the history of stuff like this. If I didn't work in tech for a living, I could totally have been a historian in some alternate reality. :-)

EDIT2: OK, false alarm. I think the Wayback Machine *does* have what I was looking for, or at least some of it. I found the SACI page. This has the "SACI Programmer's Guide", the source code, etc. Why it didn't turn up when I search earlier I'm not sure, but I decided to try again and lo and behold there it is. Maybe something to do with the issues the Internet Archive have been having after the recent security breach? Or maybe I'm just going blind. Anyway, I'm going to poke around this a bit and see what's there.

**Comment by mindcrime** [comment]
Wayback Machine has the SACI site:

https://web.archive.org/web/20040812175125/http://www.lti.pcs.usp.br/saci/

**Comment by rbordini** [comment]
Hi Philip,
Jomi deprecated SACI decades ago, interesting you still get to hear about it

**Comment by jomifred** [comment, marked as answer]
Wow, good memories! 

SACI dates from a time where we had very few tools to program agents, so I did mine 🙂. Later, JADE arrived and replaced it with a lot of advantages, so SACI was deprecated. Anyway, it was the first communication/distribution infrastructure for Jason.

I found a [paper](https://jomifred.github.io/pubs/2000/Hubner-iberamia2000.pdf) from 2000 (!!!!) that presents SACI. It is written in Portuguese, my English at the time was far below basic level. Today, translation tools can help reading it… but just for curiosity and historians :-), since it is really outdated in terms of state of the art.

**Reply by mindcrime** [reply]
Just for fun, I took a stab at having Google Gemini translate it. I put the translated doc up in my repo, but I'll definitely take it down if you don't want it "out there" Jomi. If anybody wants to take a look, here's the link:

https://github.com/mindcrime/jason-tutorials/blob/main/docs/SACI%20Doc%20Translated%20by%20LLM.pdf

**Reply by jomifred** [reply]
For me, no problem with keeping the translated version. Thanks.

**Comment by jomifred** [comment]
For historians, the [SACI Programmer's Guide](https://github.com/user-attachments/files/17475739/programmingGuide.pdf).
